File: mobile/lib/core/utils/share_url.dart
import '../config/app_config.dart';

/// Miroir de frontend/src/utils/shareUrl.js — dérive l'origine du backend
/// depuis apiBaseUrl (qui se termine par /api/v1) pour construire le lien
/// de partage d'un produit MARCHÉ. C'est CE lien qu'il faut partager,
/// jamais une URL interne à l'app : seul le backend sait générer les
/// balises Open Graph pour les robots des réseaux sociaux (voir
/// backend/src/modules/marketplace/marketplaceShare.routes.js).
String getProductShareUrl(int productId) {
  final origin = AppConfig.apiBaseUrl.replaceFirst(RegExp(r'/api/v1/?$'), '');
  return '$origin/marche/produits/$productId';
}

File: frontend/src/utils/shareUrl.js
// Origine du backend, dérivée de VITE_API_URL (ex: https://api.exemple.com/api/v1
// -> https://api.exemple.com) — c'est CE domaine qu'il faut partager pour un
// produit MARCHÉ, jamais celui du frontend : seul le backend sait générer les
// balises Open Graph pour les robots des réseaux sociaux (le frontend est une
// SPA pure, sans rendu serveur). Voir backend/src/modules/marketplace/
// marketplaceShare.routes.js.
const API_ORIGIN = (import.meta.env.VITE_API_URL || 'http://localhost:4000/api/v1').replace(/\/api\/v1\/?$/, '');

export function getProductShareUrl(productId) {
  return `${API_ORIGIN}/marche/produits/${productId}`;
}
